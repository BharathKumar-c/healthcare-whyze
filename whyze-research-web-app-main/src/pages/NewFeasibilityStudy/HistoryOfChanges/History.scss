.HistoryOfChanges-wrapper {

    h3 {
        font-weight: 700;
        font-style: normal;


        font-size: 20px;

    }

    &_Total-population {
        display: flex;
        flex-direction: row;

        p {
            padding-left: 1rem;

            font-weight: 400;
            font-size: 16px;
            line-height: 140%;
        }
    }

    &_Initial-population {
        display: flex;
        flex-direction: row;
        gap: 20px;
        padding: 0;
        align-items: center;
        margin-bottom: 1rem;

        p {

            &:nth-child(1) {
                font-style: normal;
                font-weight: 400;
                font-size: 12px;
                text-transform: capitalize;
                margin: 0;
            }

            &:nth-child(2) {
                font-weight: 500;
                font-size: 16px;
                line-height: 5px;
                text-transform: capitalize;
                margin: 0;
                margin-left: 1rem;
            }
        }



    }
}