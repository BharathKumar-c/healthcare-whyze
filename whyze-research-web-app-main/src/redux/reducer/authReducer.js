import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
