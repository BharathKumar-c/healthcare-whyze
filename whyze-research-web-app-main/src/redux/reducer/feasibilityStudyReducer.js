import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
