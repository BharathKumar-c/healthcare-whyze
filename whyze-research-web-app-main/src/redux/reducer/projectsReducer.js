import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  projects: [],
  projectId: '',
  clinicalTrial: '',
  sponsor: '',
  description: '',
  measurement: 'Metric',
  standards: '',
  segmentId: [],
};

const projectSlice = createSlice({
  name: 'projectAction',
  initialState: initialState,
  reducers: {
    setProjects: (state, { payload }) => {
      state.projects = payload;
    },
    setProjectId: (state, { payload }) => {
      state.projectId = payload;
    },
    setClinicalTrial: (state, { payload }) => {
      state.clinicalTrial = payload;
    },
    setSponsor: (state, { payload }) => {
      state.sponsor = payload;
    },
    setDescription: (state, { payload }) => {
      state.description = payload;
    },
    setStandards: (state, { payload }) => {
      state.standards = payload;
    },
    setMeasurement: (state, { payload }) => {
      state.measurement = payload;
    },
    setSegmentId: (state, { payload }) => {
      state.segmentId = payload;
    },
    setProjectSettings: (state, { payload }) => {
      state.projectId = payload.projectId;
      state.segmentId = payload.segmentId;
      state.clinicalTrial = payload.clinicalTrial;
      state.sponsor = payload.sponsor;
      state.description = payload.description;
      state.measurement = payload.measurement;
      state.standards = payload.standards;
    },
    clearSettings: (state) => {
      state.clinicalTrial = '';
      state.sponsor = '';
      state.description = '';
      state.measurement = 'Metric';
      state.standards = '';
      state.projectId = '';
      state.segmentId = [];
    },
  },
});

export const {
  setProjects,
  setProjectId,
  setProjectSettings,
  setClinicalTrial,
  setDescription,
  setSponsor,
  setMeasurement,
  setStandards,
  setSegmentId,
  clearSettings,
} = projectSlice.actions;

export default projectSlice.reducer;
