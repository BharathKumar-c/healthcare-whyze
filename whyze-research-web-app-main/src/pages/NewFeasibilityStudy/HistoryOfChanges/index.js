import React, { useEffect, useState } from 'react';
import { Node, PopulationVector } from '../../../assets';
import PropTypes from 'prop-types';
import './History.scss';

import { useSelector } from 'react-redux';
import { Tree } from '../../../components/hoc';
import { totalPatientFormatter } from '../../../utils/calculations';
import {
  feasibilityHistoryTreeFormater,
  locationHistoryTreeFormatter,
} from '../../../utils/Formatter';

const HistoryOfChanges = (props) => {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );

  const [historyTree, setHistoryTree] = useState([]);

  useEffect(() => {
    const selectedData = props.isLocation
      ? segments[activeSegment.index]?.selectedCountries
      : segments[activeSegment.index]?.feasibilityStudy;

    const formattedHistory = props.isLocation
      ? locationHistoryTreeFormatter(selectedData)
      : feasibilityHistoryTreeFormater(selectedData);

    const filteredSelectedData =
      props.isLocation && formattedHistory[0]?.children?.length === 0;
    setHistoryTree(filteredSelectedData ? [] : formattedHistory);
  }, [props.isLocation, segments, activeSegment]);

  return (
    <div className="HistoryOfChanges-wrapper">
      <h3>History of changes</h3>
      <div className="HistoryOfChanges-wrapper_Total-population">
        <PopulationVector />
        <p>Initial population : {totalPatientFormatter(props.allcount)}</p>
      </div>
      {props.isLocation && (
        <div className="HistoryOfChanges-wrapper_Total-population">
          <PopulationVector />
          <p>
            Feasibility segment:{' '}
            {totalPatientFormatter(props.feasibilityStudyCount)}
          </p>
        </div>
      )}

      {historyTree.map((hData, index) => {
        return (
          <div key={`${index}_${hData?.title}`}>
            <div className="HistoryOfChanges-wrapper_Initial-population">
              <p>{hData?.title}</p>
            </div>

            <div>
              {hData?.children?.length > 0 && (
                <Tree
                  treeData={hData?.children}
                  switcherIcon={<Node />}
                  showLine
                />
              )}
            </div>
          </div>
        );
      })}
    </div>
  );
};
HistoryOfChanges.propTypes = {
  allcount: PropTypes.number,
  feasibilityStudyCount: PropTypes.number,
  isLocation: PropTypes.bool,
};
HistoryOfChanges.defaultProps = {
  allcount: 0,
  feasibilityStudyCount: 0,
  isLocation: false,
};
export default HistoryOfChanges;
