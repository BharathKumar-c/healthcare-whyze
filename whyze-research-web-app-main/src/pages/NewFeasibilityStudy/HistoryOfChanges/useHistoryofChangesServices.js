import { useCallback, useState } from 'react';
import { useDispatch } from 'react-redux';
import { feasibilityStudyApiUrls } from '../../../constant/ApiEndPoints';
import ApiUtil from '../../../utils/ApiUtils';
import { debounce } from '../../../utils/CommonFunction';
import { setLoading } from '../../../redux/reducer/authReducer';

const useHistoryofChangesServices = () => {
  const [allcount, setAllCount] = useState(0);
  const dispatch = useDispatch();

  const countallValues = () => {
    ApiUtil.getData(`${feasibilityStudyApiUrls.initialPopulation}`)
      .then((res) => {
        setAllCount(res?.data?.count);
        dispatch(setLoading(false));
      })
      .catch((err) => {
        dispatch(setLoading(false));
        console.log(err);
      });
  };
  const optimusValue = useCallback(debounce(countallValues, 300), []);
  return {
    allcount,
    optimusValue,
  };
};

export default useHistoryofChangesServices;
