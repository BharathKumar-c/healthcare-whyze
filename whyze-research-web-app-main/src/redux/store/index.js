import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
