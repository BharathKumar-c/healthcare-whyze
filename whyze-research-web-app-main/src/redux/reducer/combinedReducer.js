import { combineReducers } from 'redux';
import feasibilityStudy from './feasibilityStudyReducer';
import project from './projectsReducer';
import auth from './authReducer';

export default combineReducers({
  feasibilityStudyReducer: feasibilityStudy,
  authReducer: auth,
  projectReducer: project,
});
