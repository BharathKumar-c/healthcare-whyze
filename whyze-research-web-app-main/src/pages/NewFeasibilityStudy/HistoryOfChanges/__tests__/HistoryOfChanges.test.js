import React from 'react';
import { act, render, waitFor, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import store from '../../../../redux/store';
import HistoryOfChanges from '..';
const mockedUsedNavigate = jest.fn();
import ApiUtil from '../../../../utils/ApiUtils';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('axios');
jest.mock('../../../../utils/ApiUtils');

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}

afterEach(() => {
  jest.clearAllMocks();
});

const caseStudy = [
  {
    name: 'Indication',
    case: 'Indication',
    condition: 'Inclusion',
    caseName: 'T',
    child: [
      {
        id: '9c663fb3',
        condition: 'diagnosed',
        operation: 'And',
        child: [],
        diagnosed: {
          period: 'day',
          value: '22',
        },
        count: 3981,
      },
    ],
  },
  {
    name: 'Demographics',
    case: 'Demographics',
    condition: 'Inclusion',
    caseName: 'T',
    child: [
      {
        id: '28133c7b',
        condition: 'Gender',
        field: 'gender',
        gender: 'Male',
        count: 3147,
      },
      {
        id: '7f565179',
        condition: 'Age range',
        field: 'age',
        ageRange: {
          min: '1',
          max: '10',
        },
        count: 0,
      },
    ],
  },
];

describe('Test cases for History Of Changes', () => {
  test('Create the snapshots for History Of Changes', () => {
    const { container } = renderWithContext(<HistoryOfChanges />);

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy = caseStudy;
      store.dispatch(setSegments(segments));
    });

    expect(container).toMatchSnapshot();
  });
  test('Render check', () => {
    const { container } = renderWithContext(<HistoryOfChanges />);

    const historyText = screen.getByText('History Of Changes');
    expect(historyText).toBeInTheDocument();
    const totalpopulation = container.getElementsByClassName(
      'HistoryOfChanges-wrapper_Total-population',
    );
    expect(totalpopulation[0]).toBeInTheDocument();
    const initialPopulation = container.getElementsByClassName(
      'HistoryOfChanges-wrapper_Initial-population',
    );
    expect(initialPopulation[0]).toBeInTheDocument();
  });

  test('get count api', async () => {
    renderWithContext(<HistoryOfChanges />);
    ApiUtil.getData.mockResolvedValue({
      data: {
        count: 3970,
      },
    });

    await waitFor(() => {
      expect(ApiUtil.getData).toHaveBeenCalledTimes(1);
    });
  });
});
