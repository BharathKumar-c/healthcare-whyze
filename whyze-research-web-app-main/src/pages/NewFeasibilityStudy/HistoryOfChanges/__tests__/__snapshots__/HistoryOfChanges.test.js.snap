// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for History Of Changes Create the snapshots for History Of Changes 1`] = `
<div>
  <div
    class="HistoryOfChanges-wrapper"
  >
    <h3>
      History Of Changes
    </h3>
    <div
      class="HistoryOfChanges-wrapper_Total-population"
    >
      <svg>
        PopulationVector.svg
      </svg>
      <p>
        Initial population : 
        0
      </p>
    </div>
    <div>
      <div
        class="HistoryOfChanges-wrapper_Initial-population"
      >
        <p>
          Indication
        </p>
      </div>
      <div>
        <div
          class="ant-tree ant-tree-show-line"
          role="tree"
        >
          <div>
            <input
              aria-label="for screen reader"
              style="width: 0px; height: 0px; display: flex; overflow: hidden; opacity: 0; border: 0px; padding: 0px; margin: 0px;"
              tabindex="0"
              value=""
            />
          </div>
          <div
            aria-hidden="true"
            class="ant-tree-treenode"
            style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
          >
            <div
              class="ant-tree-indent"
            >
              <div
                class="ant-tree-indent-unit"
              />
            </div>
          </div>
          <div
            class="ant-tree-list"
            style="position: relative;"
          >
            <div
              class="ant-tree-list-holder"
            >
              <div>
                <div
                  class="ant-tree-list-holder-inner"
                  style="display: flex; flex-direction: column;"
                >
                  <div
                    aria-grabbed="false"
                    class="ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-leaf-last"
                    draggable="false"
                  >
                    <span
                      aria-hidden="true"
                      class="ant-tree-indent"
                    />
                    <span
                      class="ant-tree-switcher ant-tree-switcher_open"
                    >
                      <svg
                        class="ant-tree-switcher-icon"
                      >
                        Node.svg
                      </svg>
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-open"
                      title="T"
                    >
                      <span
                        class="ant-tree-iconEle ant-tree-icon__customize"
                      >
                        <div
                          class="icons-wrapper"
                        >
                          <span>
                            0
                          </span>
                          <button
                            aria-checked="false"
                            class="ant-switch ant-switch-small"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </span>
                      <span
                        class="ant-tree-title"
                      >
                        T
                      </span>
                    </span>
                  </div>
                  <div
                    aria-grabbed="false"
                    class="ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-leaf-last"
                    draggable="false"
                  >
                    <span
                      aria-hidden="true"
                      class="ant-tree-indent"
                    >
                      <span
                        class="ant-tree-indent-unit ant-tree-indent-unit-start ant-tree-indent-unit-end"
                      />
                    </span>
                    <span
                      class="ant-tree-switcher ant-tree-switcher-noop"
                    >
                      <span
                        class="ant-tree-switcher-leaf-line"
                      />
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
                      title="diagnosed"
                    >
                      <span
                        class="ant-tree-iconEle ant-tree-icon__customize"
                      >
                        <div
                          class="icons-wrapper"
                        >
                          <span>
                            4K
                          </span>
                          <button
                            aria-checked="false"
                            class="ant-switch ant-switch-small"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </span>
                      <span
                        class="ant-tree-title"
                      >
                        diagnosed
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="HistoryOfChanges-wrapper_Initial-population"
      >
        <p>
          Demographics
        </p>
      </div>
      <div>
        <div
          class="ant-tree ant-tree-show-line"
          role="tree"
        >
          <div>
            <input
              aria-label="for screen reader"
              style="width: 0px; height: 0px; display: flex; overflow: hidden; opacity: 0; border: 0px; padding: 0px; margin: 0px;"
              tabindex="0"
              value=""
            />
          </div>
          <div
            aria-hidden="true"
            class="ant-tree-treenode"
            style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
          >
            <div
              class="ant-tree-indent"
            >
              <div
                class="ant-tree-indent-unit"
              />
            </div>
          </div>
          <div
            class="ant-tree-list"
            style="position: relative;"
          >
            <div
              class="ant-tree-list-holder"
            >
              <div>
                <div
                  class="ant-tree-list-holder-inner"
                  style="display: flex; flex-direction: column;"
                >
                  <div
                    aria-grabbed="false"
                    class="ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-leaf-last"
                    draggable="false"
                  >
                    <span
                      aria-hidden="true"
                      class="ant-tree-indent"
                    />
                    <span
                      class="ant-tree-switcher ant-tree-switcher_open"
                    >
                      <svg
                        class="ant-tree-switcher-icon"
                      >
                        Node.svg
                      </svg>
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-open"
                      title="T"
                    >
                      <span
                        class="ant-tree-iconEle ant-tree-icon__customize"
                      >
                        <div
                          class="icons-wrapper"
                        >
                          <span>
                            0
                          </span>
                          <button
                            aria-checked="false"
                            class="ant-switch ant-switch-small"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </span>
                      <span
                        class="ant-tree-title"
                      >
                        T
                      </span>
                    </span>
                  </div>
                  <div
                    aria-grabbed="false"
                    class="ant-tree-treenode ant-tree-treenode-switcher-close"
                    draggable="false"
                  >
                    <span
                      aria-hidden="true"
                      class="ant-tree-indent"
                    >
                      <span
                        class="ant-tree-indent-unit ant-tree-indent-unit-start ant-tree-indent-unit-end"
                      />
                    </span>
                    <span
                      class="ant-tree-switcher ant-tree-switcher-noop"
                    >
                      <span
                        class="ant-tree-switcher-leaf-line"
                      />
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
                      title="Gender"
                    >
                      <span
                        class="ant-tree-iconEle ant-tree-icon__customize"
                      >
                        <div
                          class="icons-wrapper"
                        >
                          <span>
                            3.1K
                          </span>
                          <button
                            aria-checked="false"
                            class="ant-switch ant-switch-small"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </span>
                      <span
                        class="ant-tree-title"
                      >
                        Gender
                      </span>
                    </span>
                  </div>
                  <div
                    aria-grabbed="false"
                    class="ant-tree-treenode ant-tree-treenode-switcher-close ant-tree-treenode-leaf-last"
                    draggable="false"
                  >
                    <span
                      aria-hidden="true"
                      class="ant-tree-indent"
                    >
                      <span
                        class="ant-tree-indent-unit ant-tree-indent-unit-start ant-tree-indent-unit-end"
                      />
                    </span>
                    <span
                      class="ant-tree-switcher ant-tree-switcher-noop"
                    >
                      <span
                        class="ant-tree-switcher-leaf-line"
                      />
                    </span>
                    <span
                      class="ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal"
                      title="Age range"
                    >
                      <span
                        class="ant-tree-iconEle ant-tree-icon__customize"
                      >
                        <div
                          class="icons-wrapper"
                        >
                          <span>
                            0
                          </span>
                          <button
                            aria-checked="false"
                            class="ant-switch ant-switch-small"
                            role="switch"
                            type="button"
                          >
                            <div
                              class="ant-switch-handle"
                            />
                            <span
                              class="ant-switch-inner"
                            />
                          </button>
                        </div>
                      </span>
                      <span
                        class="ant-tree-title"
                      >
                        Age range
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
